ksqldb-udf/src/main/java/io/confluent/ksql/function/udaf/Udaf.java

```diff
@@ -63,4 +67,33 @@
    * @return new value of current record
    */
   O map(A agg);
+
+  /**
+   * Some UDAFs can operate on any type.  In that case, the UDAF needs to be aware of the column
+   * type being used.  This method is called once when the UDAF is being created.
+   * Implementors may need to hold on to argument types or compute some other state to be re-used
+   * in methods like `aggregate` or `merge`.
+   * @param argTypeList The list of SqlArgument that this UDAF will receive as input.
+   */
+  default void initializeTypeArguments(List<SqlArgument> argTypeList) { }
+
+  /**
+   * Most UDAFs advertise their aggregate type statically via the Java type signature or
+   * annotations.
+   * For polymorphic UDAFs, implement this method to return the aggregate SQL Type.
+   * @return The aggregate SQL Type of type `A` returned by `aggregate` and `merge`.
+   */
+  default SqlType getAggregateSqlType() {
```

Comment:
Make this and the getReturnSqlType to return Optional<SqlType>. The ksql code is trying to use Optional more often to prevent NPE later these methods are called in the future and null is not handled.

Reply:
Done and done!